templates/api-deployment.yaml

```diff
@@ -39,6 +39,9 @@ spec:
         component: api
         stage: {{ .Values.stage | quote }}
     spec:
+    {{- if .Values.customTemplateSpec }}
+      {{ .Values.customTemplateSpec }}
+    {{- end }}
```

Comment:
Having reflected on this for a bit I'm not sure it's a good idea to provide this much configurability directly to helm chart users. I think having more scoped config is a safer approach. This gives a lot of scope to get things wrong. 

So for the nodeSelector changes, I thin maybe a nodeSelector property per service might be better?

LMK if you want a quick call to discuss the options!